javascript/ql/lib/semmle/javascript/security/dataflow/UnsafeShellCommandConstructionCustomizations.qll

```diff
@@ -166,6 +166,11 @@ module UnsafeShellCommandConstruction {
         .asExpr()
         .(BooleanLiteral)
         .getValue() = "true"
+    or
+    exists(API::Node node |
+      node.asSink() = sys.getOptionsArg() and
+      node.getMember("shell").getAValueReachingSink().mayHaveBooleanValue(true)
+    )
   }
```

Comment:
Why is the old bit not covered by the new bit (so that we could have only the new bit)?
Do you happen to have a concrete piece of code that is matched by the old and not the new?

Reply:
We are not guarantee that all `SystemCommandExecution` have a corresponding `API::Node`.  
I think there are `API::Node` for all of them right now, but if we model more `SystemCommandExecution` they might not all have `API::Node`s.

Reply:
Right, I can accept that. I might add a comment before line 162 saying something like "In case `sys.getOptionsArg` is not a sink for any API node".